src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -0,0 +1,152 @@
+        }
+
+        public int GetCurrentActiveStatementPosition([In] uint id, [MarshalAs(UnmanagedType.LPArray), Out] TextSpan[] ptsNewPosition)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetCurrentActiveStatementPosition(id, ptsNewPosition) ?? VSConstants.E_FAIL;
+        }
+
+        public int GetCurrentExceptionSpanPosition([In] uint id, [MarshalAs(UnmanagedType.LPArray), Out] TextSpan[] ptsNewPosition)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetCurrentExceptionSpanPosition(id, ptsNewPosition) ?? VSConstants.E_FAIL;
+        }
+
+        public int GetENCBuildState([MarshalAs(UnmanagedType.LPArray), Out] ENC_BUILD_STATE[] pENCBuildState)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetENCBuildState(pENCBuildState) ?? VSConstants.E_FAIL;
+        }
+
+        public int GetExceptionSpanCount([Out] out uint pcExceptionSpan)
+        {
+            pcExceptionSpan = default(uint);
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetExceptionSpanCount(out pcExceptionSpan) ?? VSConstants.E_FAIL;
+        }
+
+        public int GetExceptionSpans([In] uint celt, [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 0), Out] ENC_EXCEPTION_SPAN[] rgelt, [In, Out] ref uint pceltFetched)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetExceptionSpans(celt, rgelt, ref pceltFetched) ?? VSConstants.E_FAIL;
+        }
+
+        public int GetPEBuildTimeStamp([MarshalAs(UnmanagedType.LPArray), Out] OLE.Interop.FILETIME[] pTimeStamp)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetPEBuildTimeStamp(pTimeStamp) ?? VSConstants.E_NOTIMPL;
+        }
+
+        public int GetPEidentity([MarshalAs(UnmanagedType.LPArray), Out] Guid[] pMVID, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.BStr), Out] string[] pbstrPEName)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.GetPEidentity(pMVID, pbstrPEName) ?? VSConstants.E_FAIL;
+        }
+
+        public int StartDebuggingPE()
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.StartDebuggingPE() ?? VSConstants.S_OK;
+        }
+
+        public int StopDebuggingPE()
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.StopDebuggingPE() ?? VSConstants.S_OK;
+        }
+        #endregion
+
+        #region IVsENCRebuildableProjectCfg4 Implementation
+        public int HasCustomMetadataEmitter([MarshalAs(UnmanagedType.VariantBool)] out bool value)
+        {
+            value = true;
```

Comment:
Are we sure that's the correct behavior? Returning true here if we don't have a language service?